### ISSUE TITLE
H5Tinit generation rules broken with multi-config CMake projects

### ISSUE THREAD

**Issue by jdumas** [HDFGroup/hdf5#3029]
**Describe the bug**

Compiling HDF5 with a multi-config generator leads to compilation issues when CMake needs to be re-run. Steps to reproduce the problem:

```
git clone https://github.com/HDFGroup/hdf5.git
cd hdf5
git checkout hdf5-1_14_1-2
mkdir build
cd build
cmake -G "Ninja Multi-Config" ..
cmake --build . --config Release -j8
touch ../CMakeLists.txt
cmake --build . --config Release -j8
```

Ends up with the following error:

```
ninja: error: 'cmake_object_order_depends_target_hdf5-static_DEBUG', needed by '/work/hdf5/build/src/H5Tinit.c', missing and no known rule to make it
```

**Expected behavior**

No error when rerunning CMake.

**Platform (please complete the following information)**
 - HDF5 version: 1_14_1-2
 - OS and version: macOS. Possibly the same issue happens with the Visual Studio generator.
 - Compiler and version: AppleClang 13.1.6
 - Build system: CMake 3.25.3

**Comment by byrnHDF** [comment]
#3013, #3048, #3050, #3051 PRs should have fixed this issue.

**Comment by byrnHDF** [comment]
Also, current PR #3104 will make the issues with H5Tinit.c irrelevant.

**Comment by jdumas** [comment]
Hmm I tried again on the latest `develop` commit (281dbf4), and the problem is still the same (same steps as above to reproduce). I am looking forward to seeing #3104 merged though, that'd be a great improvement.

**Comment by derobins** [comment]
I'm going to go ahead and close this. The H5detect changes will be released in 1.14.2 in the fall.

**Comment by jdumas** [comment]
Thanks! Too bad I have to wait until the fall to be able to build a release version without requiring perl to regenerate a bunch of header files. But at least this issue is fixed now.
